src/test/mina_automation/fixture/archive.ml

```diff
@@ -0,0 +1,74 @@
+open Async
+open Mina_automation
+open Intf
+
+let logger = Logger.create ()
+
+type after_bootstrap =
+  { archive : Archive.Process.t; network_data : Network_data.t }
+
+type before_bootstrap =
+  { config : Archive.Config.t; network_data : Network_data.t }
+
+let setup_connection ~network_data =
+  let open Deferred.Let_syntax in
+  let postgres_uri = Sys.getenv_exn "POSTGRES_URI" in
+  let connection = Psql.Conn_str postgres_uri in
+  let db_name = "random_db" in
+  let%bind () = Psql.create_mina_db ~connection ~db:db_name in
+  return
+    (Archive.Config.create
+       ~config_file:(Network_data.genesis_ledger_path network_data)
+       ~postgres_uri:(postgres_uri ^ "/" ^ db_name)
+       ~server_port:3030 )
+
+module type TestCaseWithBootstrap = TestCase with type t = after_bootstrap
+
+module type TestCaseWithoutBootstrap = TestCase with type t = before_bootstrap
+
+module Make_FixtureWithBootstrap (M : TestCaseWithBootstrap) :
+  Fixture with type t = after_bootstrap = struct
+  type t = after_bootstrap
+
+  let test_case = M.test_case
+
+  let setup () =
+    let open Deferred.Or_error.Let_syntax in
+    let network_data_folder = Sys.getenv_exn "NETWORK_DATA_FOLDER" in
+    let network_data = Network_data.create network_data_folder in
+    let%bind.Deferred config = setup_connection ~network_data in
+    let executor = Archive.of_config config in
+    let%bind.Deferred archive = Archive.start executor in
+    [%log info] "Archive started successfully with " ;
+    return { archive; network_data }
+
+  let teardown t =
+    let open Deferred.Or_error.Let_syntax in
```

Comment:
If we're not guarantee to always run on a tmpfs, I think it would be beneficial for us to cleaup the tar file on disk.